test/verify/check-host-switching

```diff
@@ -0,0 +1,458 @@
+        b.wait_visible(".nav-item span[data-for='/@10.111.113.2'] button.nav-action.pf-m-secondary:not(:disabled)")
+        b.click("button:contains('Stop editing hosts')")
+        b.wait_not_visible(".nav-item span[data-for='/@localhost'] button.nav-action.pf-m-danger")
+        b.wait_not_visible(".nav-item span[data-for='/@10.111.113.2'] button.nav-action.pf-m-secondary")
+
+        b.wait_not_present(".nav-item span[data-for='/@10.111.113.2'] .nav-status")
+
+        self.add_machine(b, "10.111.113.3")
+        self.wait_dashboard_addresses(b, ["localhost", "10.111.113.3", "10.111.113.2"])
+        self.wait_dashboard_addresses(b2, ["localhost", "10.111.113.3", "10.111.113.2"])
+        self.wait_connected(b, "10.111.113.3")
+        self.wait_connected(b2, "10.111.113.3")
+
+        # Remove two
+        self.machine_remove(b, "10.111.113.2", m2)
+        self.wait_dashboard_addresses(b, ["localhost", "10.111.113.3"])
+        self.wait_dashboard_addresses(b2, ["localhost", "10.111.113.3"])
+
+        self.machine_remove(b, "10.111.113.3", m3, True)
+        self.wait_dashboard_addresses(b, ["localhost"])
+        self.wait_dashboard_addresses(b2, ["localhost"])
+
+        # Check that the two removed machines are listed in "Add Host"
+        # on both browsers
+        self.check_discovered_addresses(b, ["10.111.113.2", "10.111.113.3"])
+        self.check_discovered_addresses(b2, ["10.111.113.2", "10.111.113.3"])
+
+        # Add one back, check addresses on both browsers
+        self.add_machine(b, "10.111.113.2", True)
+        self.wait_dashboard_addresses(b, ["localhost", "10.111.113.2"])
+        self.wait_dashboard_addresses(b2, ["localhost", "10.111.113.2"])
+        self.wait_connected(b, "10.111.113.2")
+        self.check_discovered_addresses(b, ["10.111.113.3"])
+        self.check_discovered_addresses(b2, ["10.111.113.3"])
+
+        b.wait_not_present(".nav-item span[data-for='/@10.111.113.2'] .nav-status")
+
+        # And the second one, check addresses on both browsers
+        self.add_machine(b, "10.111.113.3", True)
+        self.wait_dashboard_addresses(b, ["localhost", "10.111.113.2", "10.111.113.3"])
+        self.wait_dashboard_addresses(b2, ["localhost", "10.111.113.2", "10.111.113.3"])
+        self.wait_connected(b, "10.111.113.3")
+        self.check_discovered_addresses(b, [])
+        self.check_discovered_addresses(b2, [])
+
+        b2.kill()
+
+        # Move m2 known host key to ~/.ssh, verify that it's known
+        self.machine_remove(b, "10.111.113.2", m2)
+        self.wait_dashboard_addresses(b, ["localhost", "10.111.113.3"])
+        key = m.execute("grep '10.111.113.2' /etc/ssh/ssh_known_hosts && sed -i '/10.111.113.2/d' /etc/ssh/ssh_known_hosts")
+        m.execute(
+            "mkdir -p ~admin/.ssh && echo '{0}' > ~admin/.ssh/known_hosts && chown -R admin:admin ~admin/.ssh""".format(key))
+        time.sleep(5)
```

Comment:
This at least needs a `# HACK:` with a pointer to an issue. I'm willing to land this PR with this to not unnecessarily stall it, but let's figure this out anyway. Can you please create an issue with a pointer to the log that failed?